lib/email-service.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+const path = require('path');
+const _ = require('lodash');
+const express = require('express');
+const hoganExpressStrict = require('hogan-express-strict');
+const traverse = require('express-partial-templates/lib/traverse');
+
+const Emailer = require('./emailer');
+const Formatters = require('./util/formatters');
+
+module.exports = class EmailerService {
+  constructor(options) {
+    if (!options) {
+      throw new Error('No options provided');
+    }
+
+    this.emailerOptions = _.pick(options, [
+      'host',
+      'port',
+      'ignoreTLS',
+      'auth',
+      'secure',
+      'from',
+      'replyTo'
+    ]);
+
+    const formatter = options.groupBySection ? 'bySection' : 'byField';
+    this.data = Formatters[formatter](options.data, options.steps, options.fields);
+
+    if (!this.data) {
+      throw new Error('No data provided');
+    }
+    this.customerEmail = options.customerEmail;
+    this.caseworkerEmail = options.caseworker;
+    this.subject = options.subject;
+    this.intro = options.intro || {};
+    this.outro = options.outro || {};
+    this._initApp();
+    this._initEmailer();
+    if (options.includeDate !== false) {
+      this._includeDate();
+    }
+  }
+
+  sendEmails() {
+    return Promise.all([
+      this.sendEmail(this.caseworkerEmail, 'caseworker', this.data),
+      this.sendEmail(this.customerEmail, 'customer', this.data)
+    ]);
+  }
+
+  sendEmail(to, recipient, data) {
+    return new Promise((resolve, reject) => {
+      Promise.all([
+        this._renderTemplate('formatted', recipient, data),
+        this._renderTemplate('raw', recipient, data),
+      ]).then(values => {
+        this.emailer.sendEmail(to, this.subject, values, (err, info) => {
+          if (err) {
+            // eslint-disable-next-line no-console
+            console.error('Error sending email to:', to, err);
+            return reject(err);
```

Comment:
Could we trigger an email error event? That way we could have other functions subscribe and perform actions on failures without needing to change the internals of this class